recipes/sol2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,69 @@
+from conans import ConanFile, tools, CMake
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class Sol2Conan(ConanFile):
+    name = "sol2"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ThePhD/sol2"
+    description = "C++17 Lua bindings"
+    topics = ("conan", "lua", "c++", "bindings")
+    settings = "os", "compiler"
+    license = "MIT"
+    requires = ["lua/5.3.5"]    
+    
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+    
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+        return self._cmake
+
+
+    def configure(self):
+        minimal_cpp_standard = "17"
+        try:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        except ConanInvalidConfiguration:
+            raise
+        except ConanException:
+            # FIXME: We need to handle the case when Conan doesn't know
+            # about a user defined compiler's default standard version
+            self.output.warn(
+                "Unnable to determine the default standard version of the compiler")
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+        #there is a bug in cmake install in 3.0.3, so handel this
+        if tools.Version(self.version) == "3.0.3":
```

Comment:
```suggestion
        if tools.Version(self.version) >= "3.0.3":
```

Reply:
no, this is correct, 3.0.3 needs copy, 
later versions 3.2.0, and older not, they work with cmake.install()
bot 2.x will be in a different folder, since it has a different C++ standard requirement
we could merge this logic all together into 1 conanfile.py, but I am not sure if this is a good idea

Reply:
Thanks for clarifying. In that case, yes, you can put logic related to a specific version. There are more recipes with same situation.